lard_tests/tests/end-to-end.rs

```diff
@@ -0,0 +1,115 @@
+use std::future::Future;
```

Comment:
Most files seem to use snake case, but here you use - not _, is that because its accepted that "end-to-end" is kinda one word?

Reply:
I don't know, I'm not a native English speaker, but "end-to-end" seems to be quite well-established! @intarga, should I use underscores instead? For me, it's fine either way.

Reply:
https://dictionary.cambridge.org/dictionary/english/end-to-end

It's in the dictionary, so it's definitely established. I don't know what the established practice is for converting hyphenated words to snake case is though, I personally think it would be best to either drop the hyphens entirely or replace them with underscores. My opinion is not that strongly held though, so I won't press it